Docs: add documentation for `GroupPath` (#3952)
aiidateam/aiida-core: docs/source/working_with_aiida/groups.rst

    In [2]: path = GroupPath("base1")
    In [3]: print(list(path.children))
    Out[3]: [GroupPath('base1/sub_group2', cls='<class 'aiida.orm.groups.Group'>'),
             GroupPath('base1/sub_group1', cls='<class 'aiida.orm.groups.Group'>')]

The `GroupPath` can be constructed using indexing or "divisors"::

    In [4]: path = GroupPath()
    In [5]: path["base1"] == path / "base1"
    Out[5]: True

Using the `browse` attribute, you can also construct the paths as preceding attributes.
This is useful in interactive environments, whereby available paths will be shown in the tab-completion::

    In [6]: path.browse.base1.sub_group2()
    Out[6]: GroupPath('base1/sub_group2', cls='<class 'aiida.orm.groups.Group'>')

To check the existence of a path element::

    In [7]: "base1" in path
    Out[7]: True

A group may be "virtual", in which case its label does not directly relate to a group,
or the group can be retrieved with the `get_group()` method::

    In [8]: path.is_virtual
    Out[8]: True
    In [9]: path.get_group() is None
    Out[9]: True
    In [10]: path["base1/sub_group1"].get_group() is None
    Out[10]: True
    In [11]: path["base1/sub_group1"].get_group()

>     Out[11]: <Group: "base1/sub_group1" [type core], of user user@email.com>

Reviewer: As @mbercx highlighted, these two results seen contradicting no?

Author: yep copy-paste error 😬 fixed in 9b2f222

Groups can be created and destroyed::

    In [12]: path["base1/sub_group1"].delete_group()
    In [13]: path["base1/sub_group1"].is_virtual
    Out[13]: True
    In [14]: path["base1/sub_group1"].get_or_create_group()
    Out[14]: (<Group: "base1/sub_group1" [type core], of user user@email.com>, True)
    In [15]: path["base1/sub_group1"].is_virtual
    Out[15]: False

> To traverse paths, the `children` attribute can be used, or for recursive traversal::

Reviewer 2: ```suggestion
To traverse paths, use the `children` attribute - for recursive traversal, use `walk`::
```
